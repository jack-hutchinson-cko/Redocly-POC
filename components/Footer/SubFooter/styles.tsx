import styled from 'styled-components';
import { device } from '../../css/tokens';

export const Container = styled.div`
	display: flex;
	flex-direction: column;
`;

export const GridItem = styled.div`
	width: 100%;
	display: flex;

	${device.toTablet} {
		flex-direction: column;
	}
`;

export const Title = styled.p`
	font-style: normal;
	font-weight: 500;
	font-size: 11px;
	line-height: 16px;
	color: white;
	width: 20%;

	${device.toTablet} {
		width: 100%;
		margin-bottom: 20px;
	}
`;

export const Description = styled.div`
	font-style: normal;
	font-weight: normal;
	font-size: 10px;
	line-height: 18px;
	color: var(--S20);
	margin-top: 16px;
	width: 30%;

	${device.toTablet} {
		width: 100%;
	}
`;

export const WrapperItems = styled.div`
	text-align: right;
	width: 80%;
	display: flex;
	justify-content: flex-end;

	${device.toTablet} {
		justify-content: flex-start;
		width: 100%;
		text-align: left;
	}

	${device.toMobileXL} {
		flex-direction: column;
	}
`;

export const LinkItem = styled.a`
	font-style: normal;
	font-weight: normal;
	font-size: 11px;
	line-height: 16px;
	color: var(--S20);
	text-decoration: none;
	padding-left: 40px;
	text-align: center;

	&:first-child {
		padding-left: 0;
	}

	${device.toMobileXL} {
		padding: 0 0 16px 0;
		text-align: left;
	}

	&:hover {
		text-decoration: underline;
	}
`;

export const WrapperLogos = styled.div`
	text-align: right;
	margin-top: 16px;
	width: 70%;

	a {
		padding-left: 24px;
	}

	svg * {
		fill: var(--logo-icon-fill);
	}

	${device.toLaptop} {
		a {
			padding-right: 24px;
			padding-left: 0;
		}
	}

	${device.toTablet} {
		width: 100%;
		text-align: left;
	}
`;
