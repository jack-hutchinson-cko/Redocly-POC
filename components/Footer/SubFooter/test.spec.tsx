import { shallow } from 'enzyme';
import React from 'react';
import SubFooter from './component';

class MockDate extends Date {
	constructor() {
		super('2021-05-14T11:01:58.135Z');
	}
}

describe('SubFooter component', () => {
	beforeEach(() => {
		// @ts-ignore
		global.Date = MockDate;
	});

	afterEach(() => {
		jest.resetAllMocks();
	});

	test('component matches snapshot', () => {
		expect.assertions(1);

		const wrapper = shallow(<SubFooter />);

		expect(wrapper).toMatchSnapshot();
	});
});
