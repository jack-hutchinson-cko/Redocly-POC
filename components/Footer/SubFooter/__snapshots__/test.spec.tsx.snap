// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubFooter component component matches snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.p>
      Â© 
      2021
       Checkout.com
    </styled.p>
    <styled.div>
      <styled.a
        href="/legal/terms-and-policies"
      >
        Terms & Policies
      </styled.a>
      <styled.a
        href="/legal/privacy-policy"
      >
        Privacy policy
      </styled.a>
      <styled.a
        href="/legal/certificates"
      >
        Certificates
      </styled.a>
      <styled.a
        href="/legal/cookie-policy"
      >
        Cookie Policy
      </styled.a>
      <styled.a
        href="/legal/disclaimer"
      >
        Disclaimer
      </styled.a>
      <styled.a
        href="/static/Modern_Slavery_Statement_v03.pdf"
      >
        Modern Slavery Statement
      </styled.a>
      <styled.a
        href="/static/2019_Supplier_Code_of_Conduct.pdf"
      >
        Supplier Code of Conduct
      </styled.a>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      Authorised and regulated as an electronic money institution by the UK Financial Conduct Authority under number 900816.
    </styled.div>
    <styled.div>
      <a
        href="https://www.facebook.com/checkout/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <Facebook />
      </a>
      <a
        href="https://twitter.com/checkout"
        rel="noopener noreferrer"
        target="_blank"
      >
        <Twitter />
      </a>
      <a
        href="https://www.instagram.com/checkout_com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <Instagram />
      </a>
      <a
        href="https://www.youtube.com/channel/UCC-VFhOEXmyaOekIOwZvxhg"
        rel="noopener noreferrer"
        target="_blank"
      >
        <YouTube />
      </a>
      <a
        href="https://www.glassdoor.co.uk/Overview/Working-at-Checkout-com-EI_IE837487.11,23.htm"
        rel="noopener noreferrer"
        target="_blank"
      >
        <Glassdoor />
      </a>
      <a
        href="https://www.checkout.com/blog"
        target="_blank"
      >
        <RSS />
      </a>
      <a
        href="https://www.linkedin.com/company/checkout/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <LinkedIn />
      </a>
    </styled.div>
  </styled.div>
</styled.div>
`;
